### PR TITLE
Enable ARM UWP build

### LINKED ISSUE #2945: ARM UWP build fails -- message : see reference to function template instantiation 'std::_Tree_node<std::pair
Repro steps: 

```pwsh 
$env:VCPKG_DEFAULT_TRIPLET = 'arm-uwp'
$env:CMAKE_GENERATOR = 'Visual Studio 16 2019'
$env:CMAKE_GENERATOR_PLATFORM = 'arm'
$env:CMAKE_SYSTEM_NAME = 'WindowsStore'
$env:CMAKE_SYSTEM_VERSION = '10.0'

cmake .. -DBUILD_TRANSPORT_WINHTTP=ON
cmake --build . --parallel 8
```

Sample output: 

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\ostream(284,1): error C2220: the following warning is treated as an error [D:\code\azure-sdk-for-cpp\build\sdk\template\azure-template\test\azure-template-test.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\ostream(269): message : while compiling class template member function 'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<(
int)' [D:\code\azure-sdk-for-cpp\build\sdk\template\azure-template\test\azure-template-test.vcxproj]
D:\code\azure-sdk-for-cpp\build\_deps\googletest-src\googletest\include\gtest/internal/gtest-internal.h(534): message : see reference to function template instantiation 'std::basic_ostream<char,std::char_traits<char>> &std::basic_ostream<char,std::char_traits<char>>::operator <<
(int)' being compiled [D:\code\azure-sdk-for-cpp\build\sdk\template\azure-template\test\azure-template-test.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\ostream(641): message : see reference to class template instantiation 'std::basic_ostream<char,std::char_traits<char>>' being compiled [D:\code\azure-sdk-for-cpp\build\sdk\template\a
zure-template\test\azure-template-test.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\ostream(284,1): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc [D:\code\azure-sdk-for-cpp\build\sdk\template\azure-template\test\azure
-template-test.vcxproj]
D:\code\azure-sdk-for-cpp\sdk\core\azure-core\src\http\curl\curl.cpp(591,9): error C2220: the following warning is treated as an error [D:\code\azure-sdk-for-cpp\build\sdk\core\azure-core\azure-core.vcxproj]
D:\code\azure-sdk-for-cpp\sdk\core\azure-core\src\http\curl\curl.cpp(591,9): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc [D:\code\azure-sdk-for-cpp\build\sdk\core\azure-core\azure-core.vcxproj]
  win_http_transport.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\xtree(1673,1): error C2220: the following warning is treated as an error [D:\code\azure-sdk-for-cpp\build\sdk\core\azure-core\azure-core.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\xtree(1632): message : see reference to function template instantiation 'std::_Tree_node<std::pair<const std::string,std::string>,std::_Default_allocator_traits<_Alloc>::void_pointer
> *std::_Tree<std::_Tmap_traits<_Kty,_Ty,_Pr,_Alloc,false>>::_Copy_nodes<_Moveit>(std::_Tree_node<std::pair<const std::string,std::string>,std::_Default_allocator_traits<_Alloc>::void_pointer> *,std::_Tree_node<std::pair<const std::string,std::string>,std::_Default_allocator_tra
its<_Alloc>::void_pointer> *,_Moveit)' being compiled [D:\code\azure-sdk-for-cpp\build\sdk\core\azure-core\azure-core.vcxproj]
          with
          [
              _Alloc=std::allocator<std::pair<const std::string,std::string>>,
              _Kty=std::string,
              _Ty=std::string,
              _Pr=std::less<std::string>,
              _Moveit=std::_Tree<std::_Tmap_traits<std::string,std::string,std::less<std::string>,std::allocator<std::pair<const std::string,std::string>>,false>>::_Copy_tag
          ]
```

### PR DESCRIPTION
**Pull request by gearama**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:
closes #2945 
Added configs for ARM UWP build, one api was failing
See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]

File: CMakeSettings.json [hunk 1]
```diff
@@ -1,5 +1,41 @@
...
+      "inheritEnvironments": [ "msvc_arm" ],
+      "buildRoot": "${projectDir}\\out\\build\\${name}",
+      "installRoot": "${projectDir}\\out\\install\\${name}",
+      "cmakeCommandArgs": "-DINSTALL_GTEST=OFF -DBUILD_TESTING=OFF -DBUILD_TRANSPORT_CURL=OFF -DBUILD_SAMPLES=ON -DBUILD_TRANSPORT_WINHTTP=ON -DCMAKE_SYSTEM_NAME=WindowsStore -DCMAKE_SYSTEM_VERSION=10.0",
+      "buildCommandArgs": "-v",
+      "ctestCommandArgs": "",
+      "variables": [
+        {
+          "name": "VCPKG_TARGET_TRIPLET",
+          "value": "arm-WindowsStore-static",
```
Comment (antkmsft) [inline CMakeSettings.json:34]:
```suggestion
          "value": "arm-uwp",
```

File: CMakeSettings.json [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+      "inheritEnvironments": [ "msvc_arm64" ],
+      "buildRoot": "${projectDir}\\out\\build\\${name}",
+      "installRoot": "${projectDir}\\out\\install\\${name}",
+      "cmakeCommandArgs": "-DINSTALL_GTEST=OFF -DBUILD_TESTING=OFF -DBUILD_TRANSPORT_CURL=OFF -DBUILD_SAMPLES=ON -DBUILD_TRANSPORT_WINHTTP=ON -DCMAKE_SYSTEM_NAME=WindowsStore -DCMAKE_SYSTEM_VERSION=10.0",
+      "buildCommandArgs": "-v",
+      "ctestCommandArgs": "",
+      "variables": [
+        {
+          "name": "VCPKG_TARGET_TRIPLET",
+          "value": "arm64-WindowsStore-static",
```
Comment (antkmsft) [inline CMakeSettings.json:16]:
```suggestion
          "value": "arm64-uwp",
```

**Review by antkmsft — COMMENTED** [review]
Where did you get these triplet values from?
I don't see them in https://github.com/microsoft/vcpkg/tree/master/triplets

File: sdk/core/azure-core/src/io/body_stream.cpp [hunk 3]
```diff
@@ -98,8 +98,8 @@ FileBodyStream::FileBodyStream(const std::string& filename)
   HANDLE fileHandle = INVALID_HANDLE_VALUE;
   try
   {
-#if !defined(WINAPI_PARTITION_DESKTOP) \
-    || WINAPI_PARTITION_DESKTOP // See azure/core/platform.hpp for explanation.
+#if (!defined(WINAPI_PARTITION_DESKTOP) || WINAPI_PARTITION_DESKTOP) \
+    && (!defined(WINAPI_FAMILY_APP)) // See azure/core/platform.hpp for explanation.
```
Comment (antkmsft) [inline sdk/core/azure-core/src/io/body_stream.cpp:102]: It is strange that you had to modify it. I guess if it compiles, it is ok, but just checking if this is an accidental change.

File: CMakeSettings.json (hunk 2 above)
Comment (gearama) [reply CMakeSettings.json:16]: [based on https://vcpkg.readthedocs.io/en/latest/users/triplets/ the -uwp triplets are "community" and not officially supported ( not tested by CI builds) the officially supported thing is WindowsStore

File: CMakeSettings.json (hunk 1 above)
Comment (gearama) [reply CMakeSettings.json:34]: [based on https://vcpkg.readthedocs.io/en/latest/users/triplets/ the -uwp triplets are "community" and not officially supported ( not tested by CI builds) the officially supported thing is WindowsStore

File: sdk/core/azure-core/src/io/body_stream.cpp (hunk 3 above)
Comment (gearama) [reply sdk/core/azure-core/src/io/body_stream.cpp:102]: nope, it is by design, the uwp build did not support the API there

File: CMakeSettings.json (hunk 1 above)
Comment (antkmsft) [reply CMakeSettings.json:34]: WindowsStore is may be the value for `VCPKG_CMAKE_SYSTEM_NAME`, not `VCPKG_TARGET_TRIPLET`. Vcpkg triplet is taken from https://github.com/microsoft/vcpkg/tree/master/triplets.
`arm-uwp` is official, `arm64-uwp` is community. We can compile on both.

File: CMakeSettings.json (hunk 2 above)
Comment (antkmsft) [reply CMakeSettings.json:16]: WindowsStore is may be the value for `VCPKG_CMAKE_SYSTEM_NAME`, not `VCPKG_TARGET_TRIPLET`. Vcpkg triplet is taken from https://github.com/microsoft/vcpkg/tree/master/triplets.
`arm-uwp` is official, `arm64-uwp` is community. We can compile on both.

File: sdk/core/azure-core/src/io/body_stream.cpp (hunk 3 above)
Comment (antkmsft) [reply sdk/core/azure-core/src/io/body_stream.cpp:102]: You must be mistaking. This is SDK code. We are compiling our SDK for `arm-uwp` and `arm64-uwp` for a year now, with the ifdef that was: `vcpkg install azure-core-cpp:arm-uwp azure-core-cpp:arm64-uwp`.

**Review by antkmsft — CHANGES_REQUESTED** [review]
I think there's currently misunderstanding about triplet values, and that the SDK code was compiling for `arm-uwp` and `arm64-uwp` for a year now, let's talk.

File: CMakeSettings.json (hunk 1 above)
Comment (antkmsft) [reply CMakeSettings.json:34]: And `arm64-uwp` is community for vcpkg. And SDK successfully compile there. We don't have to match vcpkg in the supported platforms exactly.

File: CMakeSettings.json (hunk 2 above)
Comment (antkmsft) [reply CMakeSettings.json:16]: And `arm64-uwp` is community for vcpkg. And SDK successfully compile there. We don't have to match vcpkg in the supported platforms exactly.

**Review by antkmsft — COMMENTED** [review]
@gearama, You should not need to modify the ifdef condition. When our SDK is being compiled as `arm-uwp` or `arm64-uwp`, there is no error. And also other conditions should've exploded as well, we have about 10-20 occurrences.

File: sdk/core/azure-core/src/io/body_stream.cpp (hunk 3 above)
Comment (gearama) [reply sdk/core/azure-core/src/io/body_stream.cpp:102]: according to the bug https://app.zenhub.com/workspaces/adp-native-team-5f68c995a9966e1e074f078d/issues/azure/azure-sdk-for-cpp/2945 it is not working , and has not been working for a while

Comment (gearama) [reply sdk/core/azure-core/src/io/body_stream.cpp:102]: after discussing with Anton this PR is actually not needed, his PR will supersede this one

**Pull request closed without merging**